packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -1484,19 +1565,20 @@ export class GarbageCollector implements IGarbageCollector {
 	 * This function identifies nodes that were referenced since last run and removes their unreferenced state, if any.
 	 * If these nodes are currently unreferenced, they will be assigned new unreferenced state by the current run.
 	 */
-	private updateStateSinceLastRun(
+	private findAllNodesReferencedBetweenGCs(
 		currentGCData: IGarbageCollectionData,
+		previousGCData: IGarbageCollectionData | undefined,
 		logger: ITelemetryLogger,
 	) {
 		// If we haven't run GC before there is nothing to do.
-		if (this.gcDataFromLastRun === undefined) {
-			return;
+		if (previousGCData === undefined) {
+			return undefined;
```

Comment:
Instead of returning undefined from this function, return currentGCData.referencedNodeIds since these are all the nodes referenced since last run. It will also make the calling code simpler.

Reply:
`currentGCData` doesn't have `referencedNodeIds` :( that's why I had it return undefined.

Reply:
We could just pass in `gcResult`

Reply:
Right. It's up to you then. Doesn't really matter. Either way, add comment at the call site to make it clear what's happening.